crates/discovery/src/source_type.rs

```diff
@@ -0,0 +1,17 @@
+use serde::{Deserialize, Serialize};
+
+/// Attribute of a device tracker that affects state when being used to track a person.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+pub enum SourceType {
```

Comment:
I don't think `SourceType` is a good name here. I suggest either `DeviceTrackerType` - or if we want to stay more 1-1 with hass, `DeviceTrackerSourceType`.

Reply:
`device_tracker_source_type::DeviceTrackerSourceType` is a mouthful but I think that just highlights that `lib.rs` should re-export most types instead of putting them under public modules anyway